### ISSUE TITLE
inconsistent behavior with PgUp PgDn and arrow keys 

### ISSUE THREAD

**Issue by nonumeros** [mawww/kakoune#3919]
This is more of a question (inquiry) — I think­— rather than a finding of anything. I was just more interested on the mouse scrolling feature and how is implemented on  kak instead. This in turn, led me to notice the following and bear with me for a second:

If I were to have kak on the tty, rather than any other terminal emulator, I'd expect PgDn and PgUp to work consistently with any given file. The word 'consistent' which I'm mentioning here, is a reference more to the behavior that one would normally get with other editors. modal or non-modal or whatnot. 

Normally, on the tty — or from what I can see — and right after scrolling all the way to the end of the file by using the `PgDn` key, the up arrow key needs to be pressed a few lines, if I wanted to have the cursor on the last line of the file, right after invoking PgDn again.. Simply put, using `PgDn` and `PgUp` exclusively, and scrolling all the way to the bottom and back to the top, won't do it. 

The display or the screen view is not in accordance with the way that other editors, whether t'd be emacs or vim, handle it. 

Sure. someone might say that it works fine with any other terminal emulator. And it does. Except with the screen view. On this instance, The cursor is usually found Only on that last line of the file and at the very top of the screen.

Is this intended? If so, it should be the same on the tty, or so I thought. But it's not.  

If I were to have `emacs -nw` or firing emacs from the tty, there's some degree of consistency, with the way that the cursor displays the view. At no point, the cursor hinders this behavior. 

Please, note that I'm not implying what the accepted behavior should be. But I always thought that the display view should prevail and not the cursor. Anyhow:

But PgDn all the way to the bottom, gives
```
[]
some lines
some lines
end of line
```

Only after using the up arrow key and invoking PgDn again then

```
[]end of line
```

with emacs
```
[]
some lines
some lines
some lines
end of line
```

with vim 
```
[]end of line regardless of PgDn or arrow keys
```

**Comment by nonumeros** [comment]
I had a copy of https://github.com/mawww/kakoune/releases/tag/v2019.01.20 that I just tried it, and unfortunately I'm not getting the same results than above.

**Comment by lenormf** [comment]
Dupe of #1517?

**Comment by nonumeros** [comment]
Between that release and now, and testing it with the same file, apparently it's pointing out to 1d9a9bc3

```
7f141e83 (HEAD, refs/bisect/good-7f141e83ce260d299ff08f724a0cdc598dc73b95) Additional NCursesUI code cleanups
```
and
```
1d9a9bc3087f45989fec3d36c800a6a23472d6c6 is the first bad commit
commit 1d9a9bc3087f45989fec3d36c800a6a23472d6c6
Author: Maxime Coste <mawww@kakoune.org>
Date:   Thu Sep 26 20:14:08 2019 +1000

    Remove unneeded NCursesUI::Window::mark_dirty and redraws
    
    A bug in client.cc was always forcing full redraws of the windows,
    leading to much more terminal output traffic than necessary.

 src/client.cc     | 2 +-
 src/ncurses_ui.cc | 8 --------
 src/ncurses_ui.hh | 1 -
 3 files changed, 1 insertion(+), 10 deletions(-)
```

**Comment by nonumeros** [comment]
I'm not even sure of the reference to wrap lines here, but I could be wrong. Tried it with and without it, and unless the screen is redrawn for each `c-d` or `c-u` or for every half page up or down, this wouldn't be the most functional/intended behavior. 

Surprisingly, at least for me, is that I yet have to get the same behavior from any of the terminal emulators I have lying around
